spyder/plugins/editor/plugin.py

```diff
@@ -2284,6 +2273,85 @@ def new(self, fname=None, editorstack=None, text=None):
+                text, enc = encoding.read(self.TEMPLATE_PATH)
+                enc_match = re.search(r'-*- coding: ?([a-z0-9A-Z\-]*) -*-',
+                                      text)
+                if enc_match:
+                    enc = enc_match.group(1)
+
+                # Initialize template variables
+                if os.name == 'nt':
+                    # Windows
+                    import ctypes
+
+                    username = encoding.to_unicode_from_fs(
+                                os.environ.get('USERNAME', '-'))
+                    GetUserNameEx = ctypes.windll.secur32.GetUserNameExW
+                    name_display = 3
+
+                    size = ctypes.pointer(ctypes.c_ulong(0))
+                    GetUserNameEx(name_display, None, size)
+
+                    name_buffer = ctypes.create_unicode_buffer(
+                                size.contents.value)
+                    GetUserNameEx(name_display, name_buffer, size)
+                    displayname = name_buffer.value.strip()
+                else:
+                    # Linux, Mac OS X
+                    import pwd
+
+                    username = encoding.to_unicode_from_fs(
+                                   os.environ.get('USER', '-'))
+                    try:
+                        pwd_struct = pwd.getpwnam(username)
+                    except KeyError:
+                        displayname = ''
+                    else:
+                        # Use first element in Gecos field as user's full name
+                        displayname = pwd_struct.pw_gecos.split(',')[0].strip()
+
+                now = datetime.now().astimezone()
+                now.replace(microsecond=0)
+                VARS = {
+                    'date': now.ctime(),
+                    'isodate': now.isoformat(),
+                    'year': now.strftime('%Y'),
+                    'month': now.strftime('%m'),
+                    'day': now.strftime('%d'),
+                    'hour': now.strftime('%H'),
+                    'minute': now.strftime('%M'),
+                    'second': now.strftime('%S'),
+                    'tzname': now.strftime('%Z'),
+                    'monthname': now.strftime('%b'),
+                    'weekday': now.strftime('%a'),
+                    'username': username,
+                    'fullname': displayname or 'n/a',
+                    'file': fname,
+                    'filename': os.path.basename(fname),
+                    'filepath': os.path.dirname(fname),
+                    'projectname': project_name or 'n/a',
+                    'projectpath': project_path or 'n/a'
+                }
+                try:
+                    text = string.Template(text).safe_substitute(VARS)
+                except Exception:
+                    pass
+            else:
+                default_content = False
+                enc = encoding.read(self.TEMPLATE_PATH)[1]
+        except (IOError, OSError):
```

Comment:
```suggestion
        except OSError:
```

`IOError` is a legacy alias of `OSError`

Reply:
Oh, indeed. I just learned something new!